src/components/picture-view-modal.vue

```diff
@@ -21,126 +28,139 @@
     <div class="interaction">
       <router-link :to="link">
         <ion-icon :icon="commentsIcon" size="large" />
-        <ion-label>{{interactivityObject.commentsCount || 0}}</ion-label>
+        <ion-label>{{ interactivityObject.commentsCount || 0 }}</ion-label>
       </router-link>
     </div>
     <div class="interaction">
       <like-button
-          icon-size="large"
-          :has-liked="hasLiked"
-          :pointer="pointer"
-          :counter="interactivityObject.likesCount || 0"
+        icon-size="large"
+        :has-liked="hasLiked"
+        :pointer="pointer"
+        :counter="interactivityObject.likesCount || 0"
       />
     </div>
-  </div> 
-</div> 
- 
+  </div>
 </template>
 
 <script lang="ts">
-import { IonLabel, IonIcon, IonImg } from '@ionic/vue';
-import { createAnimation } from '@ionic/core';
-import { chatbubblesOutline as commentsIcon, heart as likeIcon } from 'ionicons/icons';
-import Avatar from './avatar.vue';
+import { IonIcon } from "@ionic/vue";
+import { createAnimation } from "@ionic/core";
+import {
+  chatbubblesOutline as commentsIcon,
+  heart as likeIcon,
+} from "ionicons/icons";
+import Avatar from "./avatar.vue";
 import ShareButton from "./share-button.vue";
 import LikeButton from "./like-button.vue";
-import { since } from '../utils/time';
+import { since } from "../utils/time";
 import Parse from "parse";
-import { defineComponent, computed } from 'vue';
-import { useStore } from '../stores/';
-import { Model } from '@/utils/model';
+import { defineComponent, computed } from "vue";
+import { useStore } from "../stores/";
+import { Model } from "@/utils/model";
 
 export default defineComponent({
-  name: 'PictureViewModal',
-  emits: ['next'],
+  name: "PictureViewModal",
+  emits: ["next"],
   props: {
     itemId: {
       type: String,
-      required: true
+      required: true,
     },
-    indexValue:{
+    indexValue: {
       type: Number,
-    }
+    },
+    itemData: {
+      type: Array,
+    },
   },
   components: {
-    IonLabel, IonIcon,
-    IonImg,
-    Avatar, ShareButton, LikeButton, 
+    IonIcon,
+    // IonImg,
+    Avatar,
+    ShareButton,
+    LikeButton,
   },
-  setup(props: { itemId: string }) {
+  setup(props: any) {
     const store = useStore();
-  
-    return {    
-      item: computed(() => store.getters.objectsMap[props.itemId]),
+    console.log(props);
+    console.log("uri is ", props.itemId.objects[0].file.url);
+    return {
+      item: props.itemId,
+      allData: props.itemData,
+      indexOfImg: props.indexValue,
```

Comment:
props are set on the component by default, we don't have to remap them. quite the opposite, `setup` isn't always called in an ionic environment, thus we should make sure we always use `computed` or observables here to ensure the item stays reactive on changes.

but in this case, lets just skip reading `props` here and use the `itemId` and `itemData` directly (rather than remapped as `allData`)